### PR TITLE
More stardust nits

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-message/src/payload/transaction/mod.rs [hunk 1]
```diff
@@ -115,10 +71,25 @@ impl Packable for TransactionPayload {
...
-            .map_err(UnpackError::Packable)
+        if VERIFY {
+            verify_essence_unlock_blocks(&essence, &unlock_blocks).map_err(UnpackError::Packable)?;
+        }
+
+        Ok(TransactionPayload { essence, unlock_blocks })
     }
 }
+
+fn verify_essence_unlock_blocks(essence: &TransactionEssence, unlock_blocks: &UnlockBlocks) -> Result<(), Error> {
```
Comment (Thoralf-M) [inline bee-message/src/payload/transaction/mod.rs:82]: This only verifies the amount of unlock_blocks, will other validation be added later?

Comment (thibault-martinez) [reply bee-message/src/payload/transaction/mod.rs:82]: If something else is added to the TIP yes.
Do you have anything in mind ?

Comment (Thoralf-M) [reply bee-message/src/payload/transaction/mod.rs:82]:
```
Signature must contain an Ed25519 Signature.
The Signature Unlock Block must be unique, i.e. there must not be any other Signature Unlock Blocks in the Unlock Blocks field of the transaction payload with the same signature.
```
```
Unlock syntactic validation
The Reference Unlock Block at index i must have Reference < i and the unlock block at index Reference must be a Signature Unlock Block.
```
(this should also be for alias and nft unlock blocks, right? The TIP needs to be updated to add that I think

Comment (thibault-martinez) [reply bee-message/src/payload/transaction/mod.rs:82]: This is done on the unlock_block level https://github.com/iotaledger/bee/blob/shimmer-develop/bee-message/src/unlock_block/mod.rs#L109

**Review by Thoralf-M — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

File: bee-message/src/payload/transaction/essence/regular.rs [hunk 2]
```diff
@@ -26,10 +26,13 @@ pub struct RegularTransactionEssenceBuilder {
 
 impl RegularTransactionEssenceBuilder {
     /// Creates a new [`RegularTransactionEssenceBuilder`].
-    pub fn new(network_id: u64) -> Self {
+    pub fn new(network_id: u64, inputs_commitment: [u8; 32]) -> Self {
         Self {
             network_id,
-            ..Default::default()
+            inputs: Vec::new(),
+            inputs_commitment,
```
Comment (grtlr) [inline bee-message/src/payload/transaction/essence/regular.rs:33]: Why not just add `inputs_commitment` and keep the rest `..Default::default()`?

**Review by grtlr — APPROVED** [review]

File: bee-message/src/payload/transaction/essence/regular.rs (hunk 2 above)
Comment (thibault-martinez) [reply bee-message/src/payload/transaction/essence/regular.rs:33]: No real reason, personal preference 😅

**Pull request merged**